src/SecurityChecks.js

```diff
@@ -0,0 +1,122 @@
+  const options = req.config || req;
+  const schema = await options.database.loadSchema();
+  const all = await schema.getAllClasses();
+  const clpWarnings = {};
+  for (const field of all) {
+    const className = field.className;
+    const clp = field.classLevelPermissions;
+    const thisClassWarnings = clpWarnings[className] || [];
+    if (!clp) {
+      thisClassWarnings.push({
+        title: `No Class Level Permissions on ${className}`,
+        message: `Any client can create, find, count, get, update, delete, or add field on ${className}. This allows an attacker to create new objects or fieldNames without restriction and potentially flood the database. Set CLPs using Parse Dashboard.`,
+      });
+      clpWarnings[className] = thisClassWarnings;
+      continue;
+    }
+    const keys = ['find', 'count', 'get', 'create', 'update', 'delete', 'addField'];
+    for (const key of keys) {
+      const option = clp[key];
+      if (className === '_User' && key === 'create') {
+        continue;
+      }
+      if (!option || option['*']) {
+        thisClassWarnings.push({
+          title: `Unrestricted access to ${key}.`,
+          message: `Any client can ${key} on ${className}.`,
+        });
+      } else if (Object.keys(option).length != 0 && key === 'addField') {
+        thisClassWarnings.push({
+          title: `Certain users can add fields.`,
+          message: `Certain users can add fields on ${className}. This allows these users to create new fieldNames and potentially flood the schema. Set CLPs using Parse Dashboard.`,
+        });
+      }
+    }
+    clpWarnings[className] = thisClassWarnings;
+  }
+  return clpWarnings;
+}
+function checkServerConfig(req) {
+  const options = req.config || req;
+  const warnings = [];
+  if (options.allowClientClassCreation) {
+    warnings.push({
+      title: 'Client class creation allowed',
+      message:
+        'Clients are currently allowed to create new classes. This allows an attacker to create new classes without restriction and potentially flood the database. Change the Parse Server configuration to allowClientClassCreation: false.',
+    });
+  }
+  if (!options.masterKey.match('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{14,})')) {
+    warnings.push({
+      title: 'Weak masterKey.',
+      message:
+        'The masterKey set to your configuration lacks complexity and length. This could potentially allow an attacker to brute force the masterKey, exposing all the entire Parse Server.',
+    });
+  }
+  let https = false;
+  try {
+    const serverURL = url.parse(options.serverURL);
+    https = serverURL.protocol === 'https:';
+  } catch (e) {
+    /* */
+  }
+  if (!https) {
+    warnings.push({
+      title: `Parse Server served over HTTP`,
+      message:
+        'The server url is currently HTTP. This allows an attacker to listen to all traffic in-between the server and the client. Change the Parse Server configuration serverURL to HTTPS.',
+    });
+  }
```

Comment:
You should check `publicServerUrl` since `serverUrl` can be set safely to `localhost` in modern cloud system.

serverUrl is set to `localhost`, the a proxy transfer the request `https` to the localhost of the container for example.

Here in modern cloud system, developers will get false positive and also for example the `serverUrl` to localhost on prod env behind a proxy is much more safer since all calls done with Master key just stay into the localhost.